docs/core/compatibility/core-libraries/6.0/partial-byte-reads-in-streams.md

```diff
@@ -25,26 +27,41 @@ Also, when `Stream.Read` or `Stream.ReadAsync` was called with a buffer of lengt
 
 Starting in .NET 6, when `Stream.Read` or `Stream.ReadAsync` is called on one of the affected stream types with a buffer of length `N`, the operation completes when:
 
-- At least one byte has been read from the stream, or
+- At least 1 byte has been read from the stream, or
 - The underlying stream they wrap returns 0 from a call to its read, indicating no more data is available.
 
-Also, when `Stream.Read` or `Stream.ReadAsync` is called with a buffer of length 0, the operation succeeds once a call with a non-zero buffer would succeed.
+Also, when `Stream.Read` or `Stream.ReadAsync` is called with a buffer of length 0, the operation succeeds once a call with a nonzero buffer would succeed.
+
+For example, the following call to <xref:System.IO.Compression.GZipStream.Read%2A?displayProperty=nameWithType> doesn't read all of the compressed text for very long strings.
```

Comment:
@stephentoub I'm making this update in response to a request from a top/paid customer. Is there a numerical value I can use here instead of "very long"?

Reply:
This isn't about "very long strings". It's purely about whether there's data to be read. When you call Read, if the Read can satisfy at least one byte of the request, regardless of how many were requested, it will return as many as it can at that moment. Consider this example:
```C#
using System.IO;
using System;
using System.IO.Compression;
using System.Net;
using System.Net.Sockets;
using System.Threading.Tasks;

internal class Program
{
    private static async Task Main()
    {
        // Connect two sockets and wrap a stream around each
        using (Socket listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
        using (Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
        {
            listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
            listener.Listen(int.MaxValue);
            client.Connect(listener.LocalEndPoint);
            using (Socket server = listener.Accept())
            {
                var clientStream = new NetworkStream(client, ownsSocket: true);
                var serverStream = new NetworkStream(server, ownsSocket: true);

                // Create some compressed data
                var compressedData = new MemoryStream();
                using (var gz = new GZipStream(compressedData, CompressionLevel.Fastest, leaveOpen: true))
                {
                    byte[] bytes = new byte[1000];
                    new Random().NextBytes(bytes);
                    gz.Write(bytes, 0, bytes.Length);
                }

                // Trickle it from the client stream to the server
                Task sendTask = Task.Run(() =>
                {
                    foreach (byte b in compressedData.ToArray())
                    {
                        clientStream.WriteByte(b);
                    }
                    clientStream.Dispose();
                });

                // Read and decompress all the sent bytes
                byte[] buffer = new byte[1_000];
                int total = 0;
                using (var gz = new GZipStream(serverStream, CompressionMode.Decompress))
                {
                    int numRead;
                    while ((numRead = gz.Read(buffer, 0, buffer.Length)) > 0)
                    {
                        total += numRead;
                        Console.WriteLine($"Read: {numRead}");
                    }
                }
                Console.WriteLine("Total received: " + total);

                await sendTask;
            }
        }
    }
}
```
That creates and compresses a thousand random bytes. It then sends the compressed data one byte at a time from the client to the server, with the server decompressing the data with calls to Read 1000 bytes. On .NET Framework (and older .NET Core versions), that would produce this output:
```
Read: 1000
Total received: 1000
```
because even though data is available for GZipStream to return, it's forced to wait until the requested number of bytes is available.  With .NET 6+, you instead get output along the lines of this:
```
Read: 1
Read: 110
Read: 3
Read: 4
Read: 4
Read: 4
Read: 3
Read: 4
Read: 3
Read: 4
Read: 3
Read: 3
Read: 5
Read: 4
Read: 4
Read: 3
Read: 1
Read: 3
Read: 3
Read: 3
Read: 3
Read: 2
Read: 3
Read: 3
Read: 3
Read: 3
Read: 3
Read: 2
Read: 3
Read: 4
Read: 6
Read: 3
Read: 5
Read: 4
Read: 5
Read: 4
Read: 5
Read: 5
Read: 5
Read: 5
Read: 5
Read: 3
Read: 6
Read: 4
Read: 5
Read: 3
Read: 7
Read: 6
Read: 5
Read: 6
Read: 4
Read: 4
Read: 2
Read: 5
Read: 4
Read: 4
Read: 4
Read: 4
Read: 4
Read: 3
Read: 5
Read: 4
Read: 3
Read: 1
Read: 1
Read: 4
Read: 14
Read: 5
Read: 2
Read: 2
Read: 3
Read: 4
Read: 5
Read: 4
Read: 6
Read: 6
Read: 6
Read: 6
Read: 6
Read: 9
Read: 9
Read: 8
Read: 10
Read: 10
Read: 7
Read: 5
Read: 10
Read: 6
Read: 6
Read: 6
Read: 6
Read: 7
Read: 6
Read: 6
Read: 6
Read: 6
Read: 6
Read: 8
Read: 8
Read: 8
Read: 8
Read: 4
Read: 6
Read: 6
Read: 3
Read: 3
Read: 5
Read: 5
Read: 10
Read: 6
Read: 6
Read: 5
Read: 3
Read: 6
Read: 6
Read: 5
Read: 5
Read: 6
Read: 6
Read: 8
Read: 7
Read: 6
Read: 6
Read: 10
Read: 5
Read: 6
Read: 5
Read: 6
Read: 5
Read: 5
Read: 5
Read: 5
Read: 5
Read: 7
Read: 6
Read: 5
Read: 5
Read: 5
Read: 4
Read: 6
Read: 5
Read: 6
Read: 5
Read: 5
Read: 5
Read: 5
Read: 5
Read: 5
Read: 5
Read: 6
Read: 5
Read: 5
Read: 4
Read: 5
Read: 6
Read: 5
Read: 5
Read: 5
Read: 6
Read: 4
Read: 6
Read: 5
Read: 6
Read: 5
Read: 5
Read: 6
Read: 4
Read: 5
Read: 5
Read: 5
Read: 5
Read: 6
Read: 5
Read: 5
Read: 5
Read: 5
Read: 6
Read: 5
Read: 2
Total received: 1000
```
because even though the call to Read is requesting 1000 bytes, the Read call was able to successfully decompress received bytes into decompressed bytes that can be returned, and it does.